### DISCUSSION TITLE
Troubleshooting keybinds in tmux

### DISCUSSION THREAD

**Discussion by muraii** [tconbeer/harlequin#428] [Q&A, Troubleshooting, Support] [answered]
**Describe the bug**
When running `harlequin -a sqlite <dbname.db>` in a `tmux` session, neither `CTRL + ENTER` nor `CTRL + J` execute the query. Clicking the "run query" button works just fine, and both `CTRL + ENTER` and `CTRL + J` executes the query when running `harlequin` outside a `tmux` session on the same system.

**To Reproduce**
1. Create and attach new `tmux` session.
2. Execute `harlequin -a sqlite <dbname.db>`.
3. Write a valid SQL query.
4. Attempt to execute with `CTRL + ENTER` or `CTRL + J`.

**Expected behavior**
A valid SQL query is executed and results returned.

**Actual behavior**
No action appears to execute. There is no error or status message.

**Additional context**
What is the output of `harlequin --version`?
```
harlequin, version 1.12.0

Installed Adapters:
  - duckdb, version 1.12.0
  - sqlite, version 1.12.0
 ```

What database adapter are you using with Harlequin? (Default is DuckDB)
SQLite

Can you tell us more about your system?
- Shell: `zsh`
- Terminal: `Windows Terminal`
- OS of the shell: `Manjaro` in WSL2 on Windows 10
- OS of the terminal (if different from the shell): Windows 10

**Comment by tconbeer** [comment]
I can't reproduce in a slightly different environment (tmux, bash/zsh, ubuntu, wsl2 on win11). (I tried both bash and zsh)

What does your `tmux.conf` look like? Could you have bound those keys to something in tmux?

**Comment by muraii** [comment, marked as answer]
> What does your `tmux.conf` look like? Could you have bound those keys to something in tmux?

Ah yeah, I've adapted my `tmux.conf` to work with the `vim-tmux-navigator` plugin which looks for `CTRL-j` to be mapped to moving one pane down. I _knew_ there was some potential collision there but I forgot that not only do I need to specify that for the `nvim` configuration, I also did it in the `tmux` configuration.

Thing is, I don't have a configuration in `tmux.conf` nor in `vim-tmux-navigator.tmux` or related plugin stuff that captures `CTRL-ENTER`. I don't see any such configuration in my `zsh` dotfiles either, nor in the "Actions" list in Windows Terminal. But! I'll keep checking around.

Cheerio!

Edit To Add:

I dislike that issues seem to dominate discussion about tools sometimes, so I wanted to say Harlequin looks dope as hell. Working with the typical `sqlite3` command is _fine_ but Harlequin already makes it easier and nicer just in a little bit o' testing. As for keybinds, there's no way an app can account for all the kinds of cockamamie configurations found in userspace, and having the button there is a great stopgap until I probably configure `tmux` to send what Harlequin's looking for with `CTRL-ENTER`.

**Comment by tconbeer** [comment]
Windows terminal sends `ctrl+enter` as `ctrl+j`! I'm sure it's some legacy compatibility thing, but I don't think there is a control code for `ctrl+enter`
